hazelcast/src/test/java/com/hazelcast/map/impl/tx/TxnMapNearCacheInvalidationTest.java

```diff
@@ -74,6 +76,67 @@
     }
 
     @Test
+    public void txn_map_contains_newly_put_key_even_it_is_null_cached_after_addition() throws Exception {
+        final String mapName = "test";
+
+        NearCacheConfig nearCacheConfig = createNearCacheConfig(mapName);
+        nearCacheConfig.setInvalidateOnChange(true);
+
+        Config config = new Config();
+        config.getMapConfig(mapName).setNearCacheConfig(nearCacheConfig);
+
+        final HazelcastInstance instance = createHazelcastInstance(config);
+
+        final CountDownLatch waitTxnPut = new CountDownLatch(1);
+        final CountDownLatch waitNullCaching = new CountDownLatch(1);
+
+        final AtomicBoolean keyExistInTxnMap = new AtomicBoolean(false);
+
+        Thread txnThread = new Thread(new Runnable() {
+            @Override
+            public void run() {
+
+                Object result = instance.executeTransaction(new TransactionalTask<Object>() {
```

Comment:
Can't we use `TransactionalTask<Boolean>` here to save the result casting?

Reply:
fixed